### PR TITLE
demo: add localtunnel/ngrok clone as example for go

### PR DESCRIPTION
**Pull request by progrium**

### CODE REVIEW

File: demos/groktunnel/main.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+	if flag.Arg(0) != "" {
+		conn, err := net.Dial("tcp", "_new."+net.JoinHostPort(*host, *port))
+		fatal(err)
+		client := httputil.NewClientConn(conn, bufio.NewReader(conn))
+		req, err := http.NewRequest("GET", "/", nil)
+		req.Host = "_new." + net.JoinHostPort(*host, *port)
+		fatal(err)
+		client.Write(req)
+		resp, _ := client.Read(req)
```
Comment (mgood) [inline demos/groktunnel/main.go:42]: should this exit on errors?
```suggestion
		resp, err := client.Read(req)
		fatal(err)
```

File: demos/groktunnel/main.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	}
+}
+
+func main() {
+	var port = flag.String("p", "9999", "server port to use")
+	var host = flag.String("h", "vcap.me", "server hostname to use")
+
+	flag.Parse()
+
+	if flag.Arg(0) != "" {
```
Comment (mgood) [inline demos/groktunnel/main.go:34]: Did you want to provide some usage docs in the source, or maybe a README? I guess it looks like this:
```
usage:
# start server:
go run main.go
# expose port 8080:
go run main.go 8080
```

File: demos/groktunnel/main.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+			ch, err := sess.Accept()
+			fatal(err)
+			conn, err := net.Dial("tcp", "127.0.0.1:"+flag.Arg(0))
+			fatal(err)
+			go join(conn, ch)
+		}
+		return
+	}
+
+	l, err := net.Listen("tcp", net.JoinHostPort("0.0.0.0", *port))
```
Comment (mgood) [inline demos/groktunnel/main.go:56]: Should this be limited to localhost for the demo? I guess if you were to productionize this, another flag to set the host here would be helpful, but listening on all IPs by default could expose people to unwanted connections.
```suggestion
	l, err := net.Listen("tcp", net.JoinHostPort("127.0.0.1", *port))
```

**Review by mgood — APPROVED** [review]

File: demos/groktunnel/main.go (hunk 2 above)
Comment (progrium) [reply demos/groktunnel/main.go:34]: yea will add

File: demos/groktunnel/main.go (hunk 3 above)
Comment (progrium) [reply demos/groktunnel/main.go:57]: i did want this to work in cases where deployed to digitalocean or something. will add bind addr flag

**Pull request merged**